beacon-chain/rpc/validator_server.go

```diff
@@ -379,27 +375,28 @@ func (vs *ValidatorServer) validatorStatus(
 		Eth1DepositBlockNumber:    eth1BlockNumBigInt.Uint64(),
 		PositionInActivationQueue: positionInQueue,
 		DepositInclusionSlot:      depositBlockSlot,
-		ActivationEpoch:           params.BeaconConfig().FarFutureEpoch,
+		ActivationEpoch:           activationEpoch,
 	}
 }
 
 func (vs *ValidatorServer) lookupValidatorStatusFlag(validatorIdx uint64, beaconState *pbp2p.BeaconState) pb.ValidatorStatus {
 	var status pb.ValidatorStatus
 	v := beaconState.ValidatorRegistry[validatorIdx]
 	epoch := helpers.CurrentEpoch(beaconState)
+	farFutureEpoch := params.BeaconConfig().FarFutureEpoch
 
-	if v.ActivationEpoch > epoch {
+	if epoch < v.ActivationEpoch {
 		status = pb.ValidatorStatus_PENDING_ACTIVE
-	} else if epoch >= v.ActivationEpoch && epoch < v.ExitEpoch {
+	} else if v.ExitEpoch == farFutureEpoch {
 		status = pb.ValidatorStatus_ACTIVE
-	} else if v.StatusFlags == pbp2p.Validator_INITIATED_EXIT {
-		status = pb.ValidatorStatus_INITIATED_EXIT
-	} else if v.StatusFlags == pbp2p.Validator_WITHDRAWABLE {
+	} else if epoch >= v.WithdrawableEpoch {
 		status = pb.ValidatorStatus_WITHDRAWABLE
-	} else if epoch >= v.ExitEpoch && epoch >= v.SlashedEpoch {
+	} else if v.Slashed && epoch >= v.ExitEpoch {
 		status = pb.ValidatorStatus_EXITED_SLASHED
 	} else if epoch >= v.ExitEpoch {
 		status = pb.ValidatorStatus_EXITED
+	} else if v.ExitEpoch != farFutureEpoch {
```

Comment:
shouldn't we put `v.ExitEpoch != farFutureEpoch` right above `epoch >= v.ExitEpoch`? otherwise it'll never reach

Reply:
I don't think so, if we did that then `epoch >= v.ExitEpoch` would never be reached.

Reply:
Ah you are right, I wasn't thinking straight there